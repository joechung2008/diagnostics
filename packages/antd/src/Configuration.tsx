import { Table, Typography } from "antd";

const Configuration: React.FC<ConfigurationProps> = ({ config }) => {
  const data = Object.entries(config).map(([key, value]) => ({
    key,
    name: key,
    value: value,
  }));

  const columns = [
    {
      title: "Key",
      dataIndex: "name",
      key: "name",
    },
    {
      title: "Value",
      dataIndex: "value",
      key: "value",
    },
  ];

  return (
    <div>
      <Typography.Title level={2} style={{ marginBottom: 12 }}>
        Configuration
      </Typography.Title>
      <Table
        aria-label="Configuration"
        columns={columns}
        dataSource={data}
        pagination={false}
      />
    </div>
  );
};

export default Configuration;
