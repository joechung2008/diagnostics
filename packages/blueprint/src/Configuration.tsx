import { HTMLTable } from "@blueprintjs/core";

const Configuration: React.FC<ConfigurationProps> = ({ config }) => {
  const items = Object.entries(config).reduce<KeyValuePair<string>[]>(
    (previous, [key, value]) => [...previous, { key, value }],
    []
  );

  return (
    <div>
      <h3 className="bp6-heading">Configuration</h3>
      <HTMLTable aria-label="Configuration">
        <thead>
          <tr>
            <th>Key</th>
            <th>Value</th>
          </tr>
        </thead>
        <tbody>
          {items.map((item) => (
            <tr key={item.key}>
              <td>{item.key}</td>
              <td>{item.value}</td>
            </tr>
          ))}
        </tbody>
      </HTMLTable>
    </div>
  );
};

export default Configuration;
