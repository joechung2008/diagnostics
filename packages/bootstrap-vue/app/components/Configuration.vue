<template>
  <div>
    <h2>Configuration</h2>
    <BTable :items="rows" :fields="fields" small />
  </div>
</template>

<script setup lang="ts">
import type { ConfigurationProps } from "@/types/ConfigurationProps";

const props = defineProps<ConfigurationProps>();

const fields = [
  { key: "key", label: "Key" },
  { key: "value", label: "Value" }
];

const rows = computed(() =>
  Object.entries(props.config ?? {}).map(([key, value]) => ({ key, value }))
);
</script>
