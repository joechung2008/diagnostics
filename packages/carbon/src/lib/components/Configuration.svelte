<script lang="ts">
	import { DataTable } from 'carbon-components-svelte';

	const { config }: ConfigurationProps = $props();

	const headers = [
		{ key: 'key', value: 'Key', empty: false },
		{ key: 'value', value: 'Value', empty: false }
	];

	const rows = $derived(Object.entries(config).map(([key, value]) => ({ id: key, key, value })));
</script>

<h2 id="configuration">Configuration</h2>
<DataTable aria-labelledby="configuration" {headers} {rows} />
