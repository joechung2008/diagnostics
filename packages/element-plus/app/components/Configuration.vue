<template>
    <h2 id="configuration">Configuration</h2>
    <el-table :data="rows" style="width: 100%" aria-labelledby="configuration">
        <el-table-column prop="key" label="Key" show-overflow-tooltip />
        <el-table-column prop="value" label="Value" show-overflow-tooltip />
    </el-table>
</template>

<script setup lang="ts">
import type { ConfigurationProps } from "@/types"

const props = defineProps<ConfigurationProps>()

const rows = computed(() =>
    Object.entries(props.config).map(([key, value]) => ({ key, value }))
)
</script>
