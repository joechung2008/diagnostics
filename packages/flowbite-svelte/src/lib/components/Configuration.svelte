<script lang="ts">
  import {
    Heading,
    Table,
    TableHead,
    TableHeadCell,
    TableBody,
    TableBodyRow,
    TableBodyCell
  } from "flowbite-svelte";

  let { config }: ConfigurationProps = $props();

  const items: KeyValuePair<string>[] = $derived.by(() =>
    Object.entries(config).map(([key, value]) => ({
      key,
      value
    }))
  );
</script>

<div>
  <Heading tag="h2" class="leading-loose">Configuration</Heading>
  <Table aria-label="Configuration">
    <TableHead>
      <TableHeadCell>Key</TableHeadCell>
      <TableHeadCell>Value</TableHeadCell>
    </TableHead>
    <TableBody>
      {#each items as item (item.key)}
        <TableBodyRow>
          <TableBodyCell>{item.key}</TableBodyCell>
          <TableBodyCell>{item.value}</TableBodyCell>
        </TableBodyRow>
      {/each}
    </TableBody>
  </Table>
</div>
