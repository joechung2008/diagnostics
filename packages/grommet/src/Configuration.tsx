import {
  Heading,
  Table,
  TableBody,
  TableCell,
  TableHeader,
  TableRow,
} from "grommet";

const Configuration: React.FC<ConfigurationProps> = ({ config }) => {
  const items = Object.entries(config).reduce<KeyValuePair<string>[]>(
    (previous, [key, value]) => [...previous, { key, value }],
    []
  );

  return (
    <div>
      <Heading
        level={3}
        margin={{ bottom: "xsmall", horizontal: "none", top: "medium" }}
      >
        Configuration
      </Heading>
      <Table aria-label="Configuration">
        <TableHeader>
          <TableRow>
            <TableCell scope="col">Key</TableCell>
            <TableCell scope="col">Value</TableCell>
          </TableRow>
        </TableHeader>
        <TableBody>
          {items.map((item) => (
            <TableRow key={item.key}>
              <TableCell>{item.key}</TableCell>
              <TableCell>{item.value}</TableCell>
            </TableRow>
          ))}
        </TableBody>
      </Table>
    </div>
  );
};

export default Configuration;
