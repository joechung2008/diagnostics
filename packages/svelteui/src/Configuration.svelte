<script lang="ts">
  export let config: Record<string, string>

  $: items = Object.entries(config).map(([key, value]) => ({
    key,
    value
  }))
</script>

<div>
  <h2 id="configuration">Configuration</h2>
  <div class="table-container">
    <table aria-labelledby="configuration">
      <thead>
        <tr>
          <th>Key</th>
          <th>Value</th>
        </tr>
      </thead>
      <tbody>
        {#each items as item (item.key)}
          <tr>
            <td>{item.key}</td>
            <td>{item.value}</td>
          </tr>
        {/each}
      </tbody>
    </table>
  </div>
</div>

<style scoped>
  h2 {
    font-size: 1.125rem;
    font-weight: 600;
  }

  table {
    width: 100%;
    border-collapse: collapse;
    margin: 1rem 0;
  }

  table th,
  table td {
    padding: 0.75rem 1rem;
    text-align: left;
  }

  table th {
    font-weight: 600;
    font-size: 0.875rem;
    text-transform: uppercase;
    letter-spacing: 0.05em;
  }

  table td {
    font-size: 0.875rem;
  }

  table td:first-child {
    font-weight: 500;
  }

  .table-container {
    overflow-x: auto;
  }
</style>
