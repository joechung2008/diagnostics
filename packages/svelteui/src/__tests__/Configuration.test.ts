import { render } from '@testing-library/svelte'
import { describe, expect, it } from 'vitest'
import Configuration from '../Configuration.svelte'

describe('Configuration', () => {
  it('should render configuration with single key-value pair', () => {
    const { container } = render(Configuration, {
      props: {
        config: {
          'app.name': 'MyApp'
        }
      }
    })

    expect(container).toMatchSnapshot()
  })

  it('should render configuration with multiple key-value pairs', () => {
    const { container } = render(Configuration, {
      props: {
        config: {
          'app.name': 'MyApp',
          'app.version': '1.0.0',
          'database.url': 'postgresql://localhost:5432/mydb',
          'features.logging': 'true'
        }
      }
    })

    expect(container).toMatchSnapshot()
  })

  it('should render empty configuration', () => {
    const { container } = render(Configuration, {
      props: {
        config: {}
      }
    })

    expect(container).toMatchSnapshot()
  })
})
