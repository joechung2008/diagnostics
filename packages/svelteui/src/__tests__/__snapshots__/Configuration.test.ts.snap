// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Configuration > should render configuration with multiple key-value pairs 1`] = `
<div>
  <div>
    <h2
      class="svelte-1riupzq"
      id="configuration"
    >
      Configuration
    </h2>
     
    <div
      class="table-container svelte-1riupzq"
    >
      <table
        aria-labelledby="configuration"
        class="svelte-1riupzq"
      >
        <thead>
          <tr>
            <th
              class="svelte-1riupzq"
            >
              Key
            </th>
             
            <th
              class="svelte-1riupzq"
            >
              Value
            </th>
          </tr>
        </thead>
         
        <tbody>
          <tr>
            <td
              class="svelte-1riupzq"
            >
              app.name
            </td>
             
            <td
              class="svelte-1riupzq"
            >
              MyApp
            </td>
             
          </tr>
          <tr>
            <td
              class="svelte-1riupzq"
            >
              app.version
            </td>
             
            <td
              class="svelte-1riupzq"
            >
              1.0.0
            </td>
             
          </tr>
          <tr>
            <td
              class="svelte-1riupzq"
            >
              database.url
            </td>
             
            <td
              class="svelte-1riupzq"
            >
              postgresql://localhost:5432/mydb
            </td>
             
          </tr>
          <tr>
            <td
              class="svelte-1riupzq"
            >
              features.logging
            </td>
             
            <td
              class="svelte-1riupzq"
            >
              true
            </td>
             
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;

exports[`Configuration > should render configuration with single key-value pair 1`] = `
<div>
  <div>
    <h2
      class="svelte-1riupzq"
      id="configuration"
    >
      Configuration
    </h2>
     
    <div
      class="table-container svelte-1riupzq"
    >
      <table
        aria-labelledby="configuration"
        class="svelte-1riupzq"
      >
        <thead>
          <tr>
            <th
              class="svelte-1riupzq"
            >
              Key
            </th>
             
            <th
              class="svelte-1riupzq"
            >
              Value
            </th>
          </tr>
        </thead>
         
        <tbody>
          <tr>
            <td
              class="svelte-1riupzq"
            >
              app.name
            </td>
             
            <td
              class="svelte-1riupzq"
            >
              MyApp
            </td>
             
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;

exports[`Configuration > should render empty configuration 1`] = `
<div>
  <div>
    <h2
      class="svelte-1riupzq"
      id="configuration"
    >
      Configuration
    </h2>
     
    <div
      class="table-container svelte-1riupzq"
    >
      <table
        aria-labelledby="configuration"
        class="svelte-1riupzq"
      >
        <thead>
          <tr>
            <th
              class="svelte-1riupzq"
            >
              Key
            </th>
             
            <th
              class="svelte-1riupzq"
            >
              Value
            </th>
          </tr>
        </thead>
         
        <tbody />
      </table>
    </div>
  </div>
</div>
`;
